.github/workflows/status-checks.yml

```diff
@@ -79,15 +79,26 @@ jobs:
             cache: gradle
 
         - name: Initialize CodeQL
-          uses: github/codeql-action/init@v2
+          uses: github/codeql-action/init@v3
           with:
             languages: java
             queries: security-and-quality
 
         - name: Autobuild
-          uses: github/codeql-action/autobuild@v2
+          uses: github/codeql-action/autobuild@v3
 
         - name: Perform CodeQL Analysis
-          uses: github/codeql-action/analyze@v2
+          uses: github/codeql-action/analyze@v3
           with:
             category: "/language:java"
+
+    dependency-submission:
+      runs-on: ubuntu-latest
+      permissions:
+        contents: write
```

Comment:
> contents:write allows the action to create a release

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Could you explain why this action needs the ability to create a release?

Reply:
Good question. This action apparently uses the GitHub API somewhere underneath. It needs a token with content:write permissions to authenticate there.
https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28#create-a-snapshot-of-dependencies-for-a-repository

Reply:
check.